iothub/device/devdoc/API Design.md

```diff
@@ -6,297 +6,160 @@
 /// <summary>
 /// Retrieve the device properties.
 /// </summary>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
 /// <returns>The device properties.</returns>
-public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken = default);
+public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken);
 
 /// <summary>
 /// Update a single property.
 /// </summary>
 /// <param name="propertyName">Property name.</param>
 /// <param name="propertyValue">Property value.</param>
 /// <param name="componentName">The component name this property belongs to.</param>
-/// <param name="conventionHandler">A convention handler that defines the content encoding and serializer to use for the properties.</param>
+/// <param name="objectSerializer">A serializer to use for the properties.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-public Task UpdatePropertyAsync(string propertyName, object propertyValue, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+public Task UpdatePropertyAsync(string propertyName, object propertyValue, string componentName = default, ObjectSerializer objectSerializer = default, CancellationToken cancellationToken = default);
```

Comment:
updated properties and commands to use the base object serializer class, kept telemetry as "convention" that inherits from object serializer -> to highlight the difference.

- telemetry "convention" also sets some properties on message, which can technically be sone via a static method (helper).
- telemetry "convention" involves both serialization and encoding right now, which properties and commands deal with json/utf-8 by default. can this change in the future?

Reply:
This sounds correct to me since there can be multiple conventions built on a serializer.

Reply:
Change property and commands to take in a convention